cucumber-selenium-tests/src/test/java/com/skillstorm/StepDefinitions/SDTransactionHistory.java

```diff
@@ -0,0 +1,185 @@
+package com.skillstorm.StepDefinitions;
+
+import java.time.Duration;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.testng.Assert;
+
+import com.skillstorm.WebDriverSingleton;
+import com.skillstorm.PageObjects.LoginPage;
+import com.skillstorm.PageObjects.TransactionHistoryPage;
+import com.skillstorm.Utilities.Navigator;
+import com.skillstorm.Utilities.UserData.User;
+import com.skillstorm.Utilities.UserData.UserType;
+
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.*;
+
+/**
+ * Step Definitions for Transaction History feature.
+ */
+public class SDTransactionHistory {
+    WebDriver driver;
+    WebDriverWait wait;
+    Navigator navigator;
+    private TransactionHistoryPage transactionHistory;
+    private String[] transactionHistoryContent;
+
+    /**
+     * Set up the ChromeDriver and initialize TransactionHistoryPage before each test scenario tagged with @transactionhistory.
+     */
+    @Before("@transactionhistory")
+    public void setUp() {
+        driver = WebDriverSingleton.getDriver();
+        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
+        navigator = new Navigator(driver);
+        transactionHistory = new TransactionHistoryPage(driver);
+        transactionHistoryContent = new String[3]; // To store transaction details
+    }
+
+    /**
+     * Tear down the WebDriver after each test scenario tagged with @transactionhistory.
+     */
+    @After("@transactionhistory")
+    public void tearDown() {
+        WebDriverSingleton.quitDriver();
+    }
+
+    // ===================== General Step Definitions =====================
+
+    /**
+     * Navigate to the Transaction History page.
+     */
+    @Given("I am on the Transaction History page")
+    public void iAmOnTheTransactionHistoryPage() {
+        transactionHistory.getCurrentUrl();
+    }
+
+    /**
+     * Log into the web application.
+     */
+    @Given("I login into the webapp")
+    public void login() {
+        navigator.navigateTo(Navigator.PGNAME_LOGIN);
+        LoginPage loginPage = new LoginPage(driver);
+        User user = new User(UserType.PERSISTANT, "joseph.sam@gmail.com", "password1");
+        loginPage.login(user);
+    }
+
+    // ===================== Navigation Step Definitions =====================
+
+    /**
+     * Navigate to the Transactions page.
+     */
+    @Given("I start on the Transactions page")
+    public void iStartOnTheTransactionPage() {
+        transactionHistory.clickTab();
```

Comment:
This would be a great place to put a `wait.until(some element on the transaction page has loaded)` after clicking the button, instead of having the `pause(2000)` in the `clickTab()` method. This will do all of the following for you: 
1) wait for the page to load
2) assure that we get to the page before running the next scenario method
3) fail the test if we don't get to the transaction page, so we aren't running future steps we don't need to :)